File: farewell.js
var fs = require('fs');

var key = 'i hope in the next ten years there would be no other farewell letter brilliant than this one';

fs.readFile(process.argv[2], function (err, data) {
  var encrypted = data.toString().replace(/\s/g, '').split(',');
  var decrypted = []
  for (var i = 0; i < encrypted.length; ++i) {
    decrypted.push(String.fromCharCode(parseInt(encrypted[i]) ^ key.charCodeAt(i % key.length)));
  }
  process.stdout.write(decrypted.join(''));
});

File: farewell.go
package main

import  (
	"io/ioutil"
	"os"
	"regexp"
	"strings"
	"strconv"
	"fmt"
)

func main() {
	re, _ := regexp.Compile("[\n\r\t ]")
	contents, _ := ioutil.ReadFile(os.Args[1])
	encrypted := strings.Split(re.ReplaceAllString(string(contents), ""), ",")

	const Key = "i hope in the next ten years there would be no other farewell letter brilliant than this one"

	var decrypted []uint8
	for i, v := range encrypted {
		tmp, _ := strconv.Atoi(v)
		decrypted = append(decrypted, uint8(tmp ^ int(Key[i % len(Key)])))
	}

	fmt.Print(string(decrypted))
}
